src/core/tsi/alts/crypt/gsec.h

```diff
@@ -451,6 +451,28 @@ grpc_status_code gsec_aes_gcm_aead_crypter_create(const uint8_t* key,
                                                   gsec_aead_crypter** crypter,
                                                   char** error_details);
 
+/** This method has the same functionality as the HKDF_expand function from
+ *  BoringSSL, but specialized to the case of the AES-128-GCM-SHA256 and
+ *  AES-256-GCM-SHA384 ciphersuites; see https://tools.ietf.org/html/rfc5869
+ *  for specifications on the derivation function. This method computes an
+ *  output key from the |prk| secret and |info|, and writes |out_size| bytes
+ *  of the result to the |out_key| buffer.
+ *  - out_key: a buffer that the method populates with the output key, which is
+ *    owned by the caller.
+ *  - out_size: the number of bytes allocated to the |out_key| buffer.
+ *  - is_sha256: this boolean is true iff the ciphersuite employs SHA256, and
+ *    false iff it employs SHA384.
+ *  - prk: a pseudorandom key used to derive |out_key|.
+ *  - prk_size: the size of the |prk| buffer.
+ *  - info: additional context information used to derive |out_key|.
+ *  - info_size: the size of the |info| buffer.
+ *
+ *  On success, the method returns GRPC_STATUS_OK; otherwise, it returns an
+ *  appropriate error code. **/
+grpc_status_code aes_gcm_derive_data(uint8_t* out_key, size_t out_size,
+                                     bool is_sha256, const uint8_t* prk,
```

Comment:
Can we use an enum here instead of a bool? This will allow us to easily expand the function purpose in the future.

Reply:
Done.